tests/unit_tests/test_ustar_cp/test_cpdAssignUStarTh20100901.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+import matlab.engine
+import numpy as np
+
+@pytest.fixture
+def mock_stats():
+    # Create a mock Stats structure
+    class MockStats:
+        def __init__(self):
+            self.mt = np.random.rand(40)
+            self.Cp = np.random.rand(40)
+            self.b1 = np.random.rand(40)
+            self.c2 = np.random.rand(40)
+            self.cib1 = np.random.rand(40)
+            self.cic2 = np.random.rand(40)
+            self.p = np.random.rand(40)
+
+    return MockStats()
+
+def test_cpdAssignUStarTh20100901_basic(matlab_engine, mock_stats):
+    # Convert mock stats to MATLAB structure
+    stats_matlab = matlab_engine.struct()
+    for field in ['mt', 'Cp', 'b1', 'c2', 'cib1', 'cic2', 'p']:
+        setattr(stats_matlab, field, matlab.double(getattr(mock_stats, field).tolist()))
+
+    fPlot = 0
+    cSiteYr = "TestSite_2024"
+
+    # Call MATLAB function
+    CpA, nA, tW, CpW, cMode, cFailure, fSelect, sSine, FracSig, FracModeD, FracSelect = matlab_engine.cpdAssignUStarTh20100901(
+        stats_matlab, fPlot, cSiteYr, nargout=11
+    )
+
+    # Assertions
+    assert isinstance(CpA, matlab.double), "CpA should be a MATLAB double array"
+    assert isinstance(nA, matlab.double), "nA should be a MATLAB double array"
+    assert isinstance(tW, matlab.double), "tW should be a MATLAB double array"
+    assert isinstance(CpW, matlab.double), "CpW should be a MATLAB double array"
+    assert isinstance(cMode, str), "cMode should be a string"
+    assert isinstance(cFailure, str), "cFailure should be a string"
+    assert isinstance(fSelect, matlab.logical), "fSelect should be a MATLAB logical array"
+    assert isinstance(sSine, matlab.double), "sSine should be a MATLAB double array"
+    assert isinstance(FracSig, float), "FracSig should be a float"
+    assert isinstance(FracModeD, float), "FracModeD should be a float"
+    assert isinstance(FracSelect, float), "FracSelect should be a float"
+
+def test_cpdAssignUStarTh20100901_plotting(matlab_engine, mock_stats):
+    # Test with plotting enabled
+    stats_matlab = matlab_engine.struct()
```

Comment:
Doesn't work for me (Matlab runtime engine R2024a).

Reply:
Fixed